### PR TITLE
remove dependency on the legacyscheme, using self-scheme

### PR DESCRIPTION
**Pull request by stewart-yu**
**What this PR does / why we need it**:
remove kube-controller-manager's dependency on the legacyscheme, using self-scheme

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -442,6 +432,5 @@ func createRecorder(kubeClient clientset.Interface, userAgent string) record.Eve
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartLogging(glog.Infof)
 	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
-	// TODO: remove dependency on the legacyscheme
-	return eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: userAgent})
+	return eventBroadcaster.NewRecorder(kubectrlmgrconfigscheme.Scheme, v1.EventSource{Component: userAgent})
```
Comment (sttts) [inline cmd/kube-controller-manager/app/options/options.go]: This has nothing to do with component configs. Would prefer to use a local scheme here. Which types do we need in that scheme? We need corev1, don't we?

Comment (stewart-yu) [reply cmd/kube-controller-manager/app/options/options.go]: Yes. Do you means we just need `scheme` for `corev1`?

Comment (stewart-yu) [reply cmd/kube-controller-manager/app/options/options.go]: fixed

File: cmd/kube-controller-manager/app/options/options.go [hunk 2]
```diff
@@ -23,22 +23,21 @@ import (
 	"net"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	apiserveroptions "k8s.io/apiserver/pkg/server/options"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	apiserverflag "k8s.io/apiserver/pkg/util/flag"
 	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
```
Comment (sttts) [inline cmd/kube-controller-manager/app/options/options.go]: clientgokubescheme maybe?

**Pull request merged**